File: backend/src/main/java/com/domenicopisano/mtgcollectionmanager/dto/CollectionItemResponse.java
package com.domenicopisano.mtgcollectionmanager.dto;

import com.domenicopisano.mtgcollectionmanager.model.enums.CardCondition;
import com.domenicopisano.mtgcollectionmanager.model.enums.CardLanguage;

public record CollectionItemResponse(
        Long id,
        String externalCardId,
        String name,
        String expansion,
        String rarity,
        String manaCost,
        String typeLine,
        String imageUrl,
        int quantity,
        CardLanguage language,
        CardCondition condition,
        boolean foil,
        Long ownerId
) {
}

File: frontend/src/app/models/collection-item-response.ts
export interface CollectionItemResponse {
  id: number;
  externalCardId: string;
  name: string;
  expansion: string;
  rarity: string;
  manaCost: string;
  typeLine: string;
  imageUrl: string;
  quantity: number;
  language: string;
  condition: string;
  foil: boolean;
  ownerId: number;
}